### PR TITLE
[frontend] fix options filters in widgets (#5181)

### LINKED ISSUE #5181: Some filters don't support all modes and operators in widgets
## Description
In widgets, relationship_type, elementId and entity_type filters are always done with operator=eq, mode (inside the filter)=or, mode (with other filters)=and.

## Reproducible Steps
An example :

Create a widget with filters: relationship_type = uses OR related-to
The displayed graph is correct.
Same with filters: relationship_type = uses AND related-to
-> we have the same graph (the graph is done with operator 'OR')

### PR DESCRIPTION
**Pull request by Archidoit**
Only internal reviews by the Filters team

issue: https://github.com/OpenCTI-Platform/opencti/issues/5181

### CODE REVIEW

File: opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx [hunk 1]
```diff
@@ -282,6 +284,99 @@ export const filtersWithEntityType = (
...
 };
 
+// remove filter with key=entity_type and values contains 'all'
+// because in this case we want everything, so no need for filters
+export const removeAllTypesFromFilter = (inputFilters?: FilterGroup) => {
+  if (inputFilters && isFilterGroupNotEmpty(inputFilters)) {
+    const { filters, filterGroups } = inputFilters;
+    const newFilters: Filter[] = [];
+    const newFilterGroups: FilterGroup[] = [];
+    filters.forEach((filter) => {
```
Comment (jpkha) [inline opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx:294]: I think, you can use filter instead 
```
filters.filter((filter) =>!(filter.key === 'entity_type' && filter.values.includes('all')))
```

File: opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx [hunk 2]
```diff
@@ -282,6 +284,99 @@ export const filtersWithEntityType = (
...
+      if (!(filter.key === 'entity_type' && filter.values.includes('all'))) {
+        newFilters.push(filter);
+      }
+    });
```
Comment (jpkha) [inline opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx]: I think, you can use filter instead 
```
filters.filter((filter) =>!(filter.key === 'entity_type' && filter.values.includes('all')))
```

File: opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx [hunk 3]
```diff
@@ -282,6 +284,99 @@ export const filtersWithEntityType = (
...
+    filterGroups.forEach((group) => {
+      const newGroup = removeAllTypesFromFilter(group);
+      if (newGroup) filterGroups.push(newGroup);
+    });
```
Comment (jpkha) [inline opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx]: I think, you can use map instead and a filter to remove all `undefined ` if needed but I don't see a case where `removeAllTypesFromFilter ` can return undefined as we checked before ?
```
filterGroups.map((group) =>  removeAllTypesFromFilter(group))
```

**Review by jpkha — COMMENTED** [review]
Wow a lot of deleted code 👍

File: opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx (hunk 2 above)
Comment (Archidoit) [reply opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx]: fixed

File: opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx (hunk 3 above)
Comment (Archidoit) [reply opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx]: fixed

**Review by jpkha — APPROVED** [review]

File: opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx [hunk 4]
```diff
@@ -262,6 +263,7 @@ export const findFilterIndexFromKey = (
   return null;
 };
 
+// add entity types context in filters
 export const filtersWithEntityType = (
```
Comment (labo-flg) [inline opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx]: you should rename this function to a verb for consistency, for instance

```suggestion
export const injectEntityTypeFilterInFilterGroup = (
```

File: opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx [hunk 5]
```diff
@@ -282,6 +284,90 @@ export const filtersWithEntityType = (
   };
 };
 
+// remove filter with key=entity_type and values contains 'all'
+// because in this case we want everything, so no need for filters
+export const removeAllTypesFromFilter = (inputFilters?: FilterGroup) => {
```
Comment (labo-flg) [inline opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx]: I think function name is a bit misleading, we naively understand that it removes all filters on entity_types, not the 'all' special case.

```suggestion
export const removeEntityTypesAllFromFilterGroup = (inputFilters?: FilterGroup) => {
```

File: opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx [hunk 6]
```diff
@@ -282,6 +284,90 @@ export const filtersWithEntityType = (
...
+      mode: inputFilters.mode,
+      filters: newFilters,
+      filterGroups: newFilterGroups,
+    };
+  }
+  return inputFilters;
+};
+
+// construct filters and options for widgets
+export const constructFiltersAndOptions = (
```
Comment (labo-flg) [inline opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx]:
```suggestion
export const buildFiltersAndOptionsForWidgets = (
```

Comment (Archidoit) [reply opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx]: done

File: opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx (hunk 5 above)
Comment (Archidoit) [reply opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx]: done

File: opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx (hunk 4 above)
Comment (Archidoit) [reply opencti-platform/opencti-front/src/utils/filters/filtersUtils.tsx]: done

**Review by labo-flg — APPROVED** [review]

**Pull request merged**